src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -568,6 +568,7 @@ class CompositeComparisonOperatorSegment(BaseSegment):
     ),
     # This is a placeholder for other dialects.
     SimpleArrayTypeGrammar=Nothing(),
+    AutoIncrementGrammar=Sequence("AUTO_INCREMENT"),
```

Comment:
I'm not sure any other dialect supports this other than MySQL so wonder if we should just remove this and move it to MySQL? Or set this to Nothing in ANSI (and remove from other keyword files)? Other dialects do have a similar concept (`AUTOINCREMENT`/`IDENTITY` in Snowflake, `SERIAL` in Posgres, nothing in BigQuery)

Reply:
Agree - will spin that up in a separate PR. Gonna merge as-is for now as it's blocking a couple of tickets I want to clear out